baseplate/frameworks/pyramid/__init__.py

```diff
@@ -239,8 +241,10 @@ def _on_new_request(self, event: pyramid.events.ContextFound) -> None:
         )
         span.set_tag("protocol", "http")
         span.set_tag("http.url", request.url)
+        span.set_tag("http.route", request.matched_route.pattern)
         span.set_tag("http.method", request.method)
         span.set_tag("peer.ipv4", request.remote_addr)
+        span.set_tag("http.request_length", request.content_length)
```

Comment:
I think we might need to handle a case where this is `None`:

```
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/wsgiref/handlers.py", line 138, in run
    self.finish_response()
  File "/usr/local/Cellar/python@3.9/3.9.12/Frameworks/Python.framework/Versions/3.9/lib/python3.9/wsgiref/handlers.py", line 183, in finish_response
    for data in self.result:
  File "/Users/matt.terwilliger/src/baseplate.py/baseplate/frameworks/pyramid/__init__.py", line 60, in __next__
    self.span.finish()
  File "/Users/matt.terwilliger/src/baseplate.py/baseplate/__init__.py", line 644, in finish
    observer.on_finish(exc_info)
  File "/Users/matt.terwilliger/src/baseplate.py/baseplate/observers/prometheus.py", line 113, in on_finish
    self.metrics.request_size_bytes_metric(self.tags).observe(
  File "/usr/local/lib/python3.9/site-packages/prometheus_client/metrics.py", line 597, in observe
    self._sum.inc(amount)
  File "/usr/local/lib/python3.9/site-packages/prometheus_client/values.py", line 22, in inc
    self._value += amount
TypeError: unsupported operand type(s) for +=: 'float' and 'NoneType'
```

in testing with a `GET` request that has no body